pkg/core/storage/boltdb_store.go

```diff
@@ -22,23 +23,55 @@ type BoltDBStore struct {
 
 // NewBoltDBStore returns a new ready to use BoltDB storage with created bucket.
 func NewBoltDBStore(cfg dbconfig.BoltDBOptions) (*BoltDBStore, error) {
-	var opts *bbolt.Options       // should be exposed via BoltDBOptions if anything needed
+	var opts *bbolt.Options // should be exposed via BoltDBOptions if anything needed
+	initFunc := func(tx *bbolt.Tx) error {
+		_, err := tx.CreateBucketIfNotExists(Bucket)
+		if err != nil {
+			return fmt.Errorf("could not create root bucket: %w", err)
+		}
+		return nil
+	}
+
+	return newBoltDBStore(cfg, opts, true, initFunc)
+}
+
+// NewROBoltDBStore returns a new read-only BoltDB storage instance and an error
+// if DB does not exist or root bucket was not found.
+func NewROBoltDBStore(cfg dbconfig.BoltDBOptions) (*BoltDBStore, error) {
```

Comment:
Why don't we use `dbconfig.BoltDBOptions` for it?

Reply:
Because we then need to expose both via `NewBoltDBStore`, and currently we handle only `ReadOnly` from the whole set of these options, maybe some of them need additional pre-setups or checks.

Could you clarify the desired API of these two methods?

Reply:
We can have `ReadOnly` in `BoltDBOptions`, `LevelDBOptions` and `DBConfiguration` at the same time if we need to. `NewBoltDBStore` already accepts generic `BoltDBOptions`, we can have as many options there as needed without new API. And it keeps RO logic entirely contained in the `New*Store` functions.